docs/features/kubernetes/authentication.md

```diff
@@ -28,6 +28,28 @@ The providers available as server side are:
 - `localKubectlProxy`
 - `serviceAccount`
 
+### Azure
+
+The Azure server side authentication provider works by authenticating on the server with
+the Azure CLI `az login`, meaning any user that can access the backstage app will have
```

Comment:
I think the truth is a bit more complicated than "it uses `az login`" -- I'd love to get input from @goenning who added this auth provider, but in [this comment](https://github.com/backstage/backstage/pull/11299#issuecomment-1119584630) they describe some of the subtleties of that `DefaultAzureCredential` from microsoft. Technically it _can_ use `az login` if the azure CLI is available to the backstage process, but it will try getting creds from [`AZURE_*` env vars](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet) or [Azure's Managed Identity service](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.managedidentitycredential?view=azure-dotnet) first, according to [these docs](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet).

Reply:
Also maybe worth noting (though it's quite technical and specific) that azure CLI command that gets run by the `AzureCliCredential` is not `az login`, but [`az account get-access-token --output json --resource 6dae42f8-4368-4678-94ff-3960e28e3630`](https://github.com/Azure/azure-sdk-for-js/blob/eba5089bba45358b62f1d3b1b7c33972f224e812/sdk/identity/identity/src/credentials/azureCliCredential.ts#L50-L59), where that GUID is the ID of the "Azure Kubernetes Service AAD Server" used by all AAD-enabled AKS clusters.

Reply:
I've updated the text & steps, should I go in detail about those other ways to provide credentials?

Reply:
The extra details here are very nice, props.

I know you didn't actually explore this in your investigations, but unfortunately the azure CLI use case is probably not the most important one to document -- it would mainly be used when iterating on a Backstage app running locally. The env var and managed identity options are more suitable for production use cases -- I imagine that's what @goenning uses for real.

You probably don't want to document something you haven't actually tested yourself, and I don't think it's worth investing your time in testing those other cases at this point, so I'm happy resolving this conversation for now. It's a big improvement!